### ISSUE TITLE
Lineage 17 for Z20

### ISSUE THREAD

**Issue by beforesq** [gesangtome/android_device_nubia_nx627j#9]
Hello. Are we able to flash the system to our phone. Do you know when it will be done? Everyone is excited and thank you for working so hard 😁😁😁

**Comment by gesangtome** [comment]
@beforesq 
No, it's not working yet. I need more people involved. I'm not going to give up easily

**Comment by beforesq** [comment]
Thanks for telling me. I am trying to contact nubia support service from china and europe to give me the kernel. If I get it, I will tell you :)

**Comment by gesangtome** [comment]
> Thanks for telling me. I am trying to contact nubia support service from china and europe to give me the kernel. If I get it, I will tell you :)

Thank you for this meaningful help and I look forward to good news.

**Comment by beforesq** [comment]
I appreciate. The phone is pretty much dead at this category, but, even with this system (I am using the global rom ) I got 475000 on antutu and Im in TOP 10. If you manage to port lineage or MIUI 12 , which is better if you ask me, I think it will get at least 480 000 on antutu. I am trying really hard to get the kernel. Maybe you can port miui 12.

**Comment by gesangtome** [comment]
> I appreciate. The phone is pretty much dead at this category, but, even with this system (I am using the global rom ) I got 475000 on antutu and Im in TOP 10. If you manage to port lineage or MIUI 12 , which is better if you ask me, I think it will get at least 480 000 on antutu. I am trying really hard to get the kernel. Maybe you can port miui 12.

Thank you for your advice. I will not port android rom, there is a more suitable person, but that person is not me. 
If you have more information about the kernel, please let me know, it is very helpful for the current adaptation work. :)

**Comment by beforesq** [comment]
https://github.com/ztemt/NX627J_P_kernel

**Comment by beforesq** [comment]
What do you think about this one? Is it good?

**Comment by beforesq** [comment]
or this one: https://github.com/nubia-development/android_kernel_nubia_sm8150

**Comment by beforesq** [comment]
Also, this guy edited the global rom. You can ask him. He might help you: https://bbs.nubia.com/thread-2552758-1-1.html

**Comment by gesangtome** [comment]
@beforesq
 If you compare with the Qualcomm public version, you will find that the necessary drivers are missing in the kernel. I started the kernel upgrade work very long ago. Unfortunately, it cannot be started.
defconfig can be extracted via adb command line: adb pull /proc/config.gz ./

**Comment by beforesq** [comment]
I contacted the guy that released almost all the nubia kernels. I hope he will be able to help me. He even released the nubia x full kernel...If he will do it, I'll let you know.

**Comment by gesangtome** [comment]
@beforesq 
 Waiting ... :)

**Comment by beforesq** [comment]
He said that this is the kernel: https://github.com/ztemt/NX627J_P_kernel of this system: https://global.mall.nubia.com/blogs/news/nubia-z20-global-rom-flash-instructions

**Comment by gesangtome** [comment]
@beforesq 
Not useful. I think more small partners should be called in to give official feedback about the open source code.

**Comment by beforesq** [comment]
Are you trying to develop a rom with the kernel of the chinese rom? Anyway, I think nobody wants to port anything to this device. I am planning to change it. I think I got scammed :(

**Comment by beforesq** [comment]
Nubia is not going to give me anything on the email. They said they don't have any kernel. "It does not exist". I think it is impossible for developers to port anything to this device. I have no idea why they are doing this. You, the developers,are just trying to help...

**Comment by gesangtome** [comment]
@beforesq
 In my opinion, the gap between the Chinese version and the international version is nothing more than the Chinese version is pre-installed with a bunch of software and no google service components. This is not a problem. At present, the main focus should be on the kernel. Do you understand what I mean?

**Comment by beforesq** [comment]
Yeah, I understand, but I can't help you. Nobody wants to help me with the kernel. I contacted almost everyone to ask for the kernel and I get the same answer : "It does not exist."

**Comment by gesangtome** [comment]
@beforesq
It seems that nubia does not want to release the kernel source code. We can only figure it out

**Comment by beforesq** [comment]
The phone has really good specs. I scored 479000 in antutu ( #14 in the top ). But the software is horrible. I might switch to the s20 or note 10. I'll see. For now, there is no hope for this device. It was a bad choice to buy it.

**Comment by gesangtome** [comment]
@beforesq 
I think so.

**Comment by beforesq** [comment]
Hey! Look at this: https://bbs.nubia.com/thread-2587681-1-1.html
@gesangtome

**Comment by beforesq** [comment]
The creator of the miui 12 port for nubia x, created a post for nubia z20 about porting the same system to the z20. The moderators deleted his post, sady. This is his profile if you want to talk with him. I think he has more informations, since he wants to port those roms. 
https://bbs.nubia.com/space-uid-9758517.html

**Comment by deadman96385** [comment]
Gonna jump on here because it seems active, @gesangtome i messaged you on telegram but you seem to be offline for awhile there. I'm working on a different Nubia sm8150 device and was wondering if you would like to work together on a common tree. I rebuilt the Nubia Mini 5g kernel source atleast far enough for it to get boot animation still working on fixing up some of the driver's. Please look at your telegram so we can discuss things my username is same as here deadman96385.

Also what exactly is missing in the z20 kernel source from my quick look (stole a few driver's from it) it seems complete.

**Comment by gesangtome** [comment]
> The creator of the miui 12 port for nubia x, created a post for nubia z20 about porting the same system to the z20. The moderators deleted his post, sady. This is his profile if you want to talk with him. I think he has more informations, since he wants to port those roms.
> https://bbs.nubia.com/space-uid-9758517.html

I'll look it.

**Comment by beforesq** [comment]
Okay. Thanks. @gesangtome

**Comment by beforesq** [comment]
@gesangtome any progress with that guy / with the system?

**Comment by gesangtome** [comment]
@beforesq 
Recently nubia decided to open source audio_kernel, which is a good start.




---Original---

**Comment by beforesq** [comment]
@gesangtome This is really good. i am thinking to change my phone with the new S20 from Samsung. i wish you good luck, but samsung is way better than all these chinese companies. They're providing live support and monthly updates.

**Comment by beforesq** [comment]
@gesangtome 
I am trying to find a way to contact someone to give us a global rom update. Im still on android 9.. I would like 10 .. Xiaomi and Google are preparing to release android 11. This is amazing.

**Comment by beforesq** [comment]
https://forum.xda-developers.com/showthread.php?t=2196491
would this work?

**Comment by gesangtome** [comment]
I will take a look. :)




---Original---

**Comment by beforesq** [comment]
Okay,thanks. I am really trying to help , because this: https://global.mall.nubia.com/blogs/news/nubia-z20-global-rom-flash-instructions   is horrible.

**Comment by beforesq** [comment]
This guy is on android 10, take a look: https://i.imgur.com/A9rDub1.png
this is the post: https://bbs.nubia.com/thread-2590955-1-1.html
this is his profile : https://bbs.nubia.com/space-uid-9314641.html
@gesangtome

**Comment by gesangtome** [comment]
I know him, it seems that he has successfully transplanted miui12.


---Original---

**Comment by beforesq** [comment]
Really? Can you please ask him if he can give me that system? I am not able to talk on that forum because I live in Europe. Please ask him . I would love to install that system on my phone.
@gesangtome

**Comment by beforesq** [comment]
@gesangtome

**Comment by beforesq** [comment]
You would help me a lot by doing this. @gesangtome 🙃

**Comment by beforesq** [comment]
I am using this rom on my phone : https://global.mall.nubia.com/blogs/news/nubia-z20-global-rom-flash-instructions
Do you know any rom for this phone? Do we have android 10 for it? Android 9 is so old...
@gesangtome

**Comment by gesangtome** [comment]
I don't have much time, I will only continue LOS17.1 when I am free.&nbsp;In addition, I have not used the international version, I think it should be android pie.

---Original---

**Comment by beforesq** [comment]
> I don't have much time, I will only continue LOS17.1 when I am free.&nbsp;In addition, I have not used the international version, I think it should be android pie.
> […](#)
> ---Original--- From: "beforesq"<notifications@github.com&gt; Date: Fri, Jul 10, 2020 06:47 AM To: "gesangtome/android_device_nubia_nx627j"<android_device_nubia_nx627j@noreply.github.com&gt;; Cc: "Mention"<mention@noreply.github.com&gt;;"gesangtome"<gesangtome@foxmail.com&gt;; Subject: Re: [gesangtome/android_device_nubia_nx627j] Lineage 17 for Z20 (#9) I am using this rom on my phone : https://global.mall.nubia.com/blogs/news/nubia-z20-global-rom-flash-instructions Do you know any rom for this phone? Do we have android 10 for it? Android 9 is so old... @gesangtome — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Yes, Android pie with tons of bugs. The system is horrible and unusable without root and some magisk modules. Can you ask that guy that you mentioned above if he can give me or you MIUI ? I would love to use my phone with a fast OS. @gesangtome

**Comment by beforesq** [comment]
@gesangtome someone ported miui 12 for nubia z20, but i don't understand chinese. Can you please tell me what is the author saying in that picture? If it's released, can you give me the file please? I would appreciate a lot. :)

**Comment by gesangtome** [comment]
I have forwarded your message to him, and if he wishes, he will take the initiative to contact you. :)


---Original---

**Comment by beforesq** [comment]
Thank you very much.I appreciate it. Do you need my email or something?

**Comment by gesangtome** [comment]
Hello, beforesq
The news I know is that he is repairing the sensor-related things, if it is OK, he will take the initiative to contact you.


---Original---

**Comment by beforesq** [comment]
> Hello, beforesq The news I know is that he is repairing the sensor-related things, if it is OK, he will take the initiative to contact you.
> […](#)
> ---Original--- From: "beforesq"<notifications@github.com&gt; Date: Thu, Jul 16, 2020 22:31 PM To: "gesangtome/android_device_nubia_nx627j"<android_device_nubia_nx627j@noreply.github.com&gt;; Cc: "Mention"<mention@noreply.github.com&gt;;"gesangtome"<gesangtome@foxmail.com&gt;; Subject: Re: [gesangtome/android_device_nubia_nx627j] Lineage 17 for Z20 (#9) Thank you very much.I appreciate it. Do you need my email or something? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

Thanks for keeping me up to date. I appreciate and I am extremely excited to use it on my phone. I think I installed too many magisk modules and too many things to my phone because it is a little bit slow, or maybe it's the system. I have no idea, but I'll use his system as soon as he will release it. 😄

**Comment by beforesq** [comment]
Hello. Any news about the guy with MIUI 12 or you with the Android 10 ROM? I am so excited ^-^

**Comment by ghost** [comment]
Please, i want miui 12 on my phone (^o^)<3

**Comment by beforesq** [comment]
> Please, i want miui 12 on my phone (^o^)<3

https://c.mi.com/global/miuidownload/index

Download it. Gesangtome is porting LineageOS , not miui!!!

**Comment by turkascode** [comment]
Please ... I am wait Android 10 , 11.. lineage os 17.1 or MIUI 12

Android 11 for wait...
https://bbs.nubia.com/forum.php?mod=viewthread&tid=2604219&extra=page%3D1&mobile=2

**Comment by turkascode** [comment]
When lineage os 17.1 coming ?

**Comment by turkascode** [comment]
Is there any improvement?

**Comment by gesangtome** [comment]
The project is dead @turkascode @Peno3000 @beforesq

**Comment by turkascode** [comment]
Thanks Bro

25 Mar 2023 Cmt 18:06 tarihinde gesangtome ***@***.***> şunu
yazdı:
